hal/rfcs/0000-spi-overhaul.md

```diff
@@ -0,0 +1,216 @@
+# HAL overhaul : SPI
+
+## Description
+
+This updates targets the SPI HAL as the first step of a wide plan for all HAL APIs.
+
+## Motivation
+
+This work is part of the effort to bring well defined specification to APIs in `mbed-os`. Some users have shown interest in some addition to the supported features of this API ([bit ordering selection #4789](https://github.com/ARMmbed/mbed-os/issues/4789)).  
+Analysis of previous work also revealed some inconsistencies in the current SPI HAL API that this RFC aims at fixing. Those inconsistencies are listed in the next chapter.  
+Finally, `mbed-os` has grown a lot its supported target count and some of the oldest one could benefit quite significantly in maintenance cost from a factorisation of their code base.
+
+### Inconsistencies
+
+- The type used in block transfer functions vary between `(const) void*` and `(const) char*`.
+- Some method have a parameter to set the width of the symbols while it is part of the `spi_format` arguments.
```

Comment:
how it could be (data bits dictate the data type aka width) however in the current api it is separated. application buffer can have diff width than data bits set in peripheral.

Thus would rephrase this to "width in buffers could be removed and use data bits to set the data type" ?